modules/end-to-end-test/src/test/groovy/org/hawkular/inventory/rest/test/InventoryITest.groovy

```diff
@@ -0,0 +1,261 @@
+
+        /* Create a metric type */
+        response = postDeletable(path: "/hawkular/inventory/metricTypes", body: [id : responseTimeMTypeId, unit : "MILLI_SECOND"])
+        assertEquals(201, response.status)
+        assertEquals(baseURI + "/hawkular/inventory/metricTypes/$responseTimeMTypeId", response.headers.Location)
+
+        /* link pingableHostRTypeId with responseTimeMTypeId */
+        response = postDeletable(path: "/hawkular/inventory/resourceTypes/$pingableHostRTypeId/metricTypes",
+                body : [id : responseTimeMTypeId])
+        assertEquals(204, response.status)
+
+        /* add a metric */
+        response = postDeletable(path: "/hawkular/inventory/$environmentId/metrics",
+                body: [ id : responseTimeMetricId, metricTypeId : responseTimeMTypeId ]);
+        assertEquals(201, response.status)
+        assertEquals(baseURI + "/hawkular/inventory/$environmentId/metrics/$responseTimeMetricId", response.headers.Location)
+
+
+        /* add a resource */
+        response = postDeletable(path: "/hawkular/inventory/$environmentId/resources",
+            body: [ id : host1ResourceId, resourceTypeId: pingableHostRTypeId ])
+        assertEquals(201, response.status)
+        assertEquals(baseURI + "/hawkular/inventory/$environmentId/resources/$host1ResourceId", response.headers.Location)
+
+        /* add another resource */
+        response = postDeletable(path: "/hawkular/inventory/$environmentId/resources",
+            body: [ id : host2ResourceId, resourceTypeId: pingableHostRTypeId ])
+        assertEquals(201, response.status)
+        assertEquals(baseURI + "/hawkular/inventory/$environmentId/resources/$host2ResourceId", response.headers.Location)
+
+        /* link the metric to resource */
+        response = client.post(path: "/hawkular/inventory/$environmentId/resources/$host1ResourceId/metrics",
+            body: [ responseTimeMetricId ]);
+        assertEquals(204, response.status)
+
+        /* add a feed */
+        response = postDeletable(path: "/hawkular/inventory/$environmentId/feeds", body: [ id : feedId ])
+        assertEquals(201, response.status)
+        assertEquals(baseURI + "/hawkular/inventory/$environmentId/feeds/$feedId", response.headers.Location)
+
+    }
+
+    @AfterClass
+    static void deleteEverything() {
+        /* the following would delete all data of the present user. We cannot do that as long as we do not have
+         * a dedicated user for running this very single test class. */
+        // def response = client.delete(path : "/hawkular/inventory/tenant")
+        // assertEquals(204, response.status)
+
+        /* Let's delete the entities one after another in the inverted order as we created them */
+        for (int i = pathsToDelete.size() - 1; i >= 0; i--) {
+            String path = pathsToDelete.get(i);
+            def response = client.delete(path : path)
+            assertEquals(204, response.status)
+        }
+
```

Comment:
Well, the GET would basically be a check if give metric type is associated with the resource type or not.

You can get the metric type by doing `/metricTypes/{metricTypeId}`. So while it would be good to have what you suggest just to be consistent, IMHO, it is not crucial to have that endpoint defined.